### PR TITLE
Add gcc-8 to Travis.

### PR DESCRIPTION
**Pull request by marxin**

### CODE REVIEW

**Comment by marxin** [comment]
It should add GCC-8 as mentioned in https://github.com/godotengine/godot/issues/23015#issuecomment-464873733

File: .travis.yml [hunk 1]
```diff
@@ -77,6 +77,18 @@ matrix:
...
+      os: linux
+      compiler: gcc-8
+      addons:
+        apt:
+          packages:
+            - gcc-8
+            - *linux_deps
+      env:
+        - CC=gcc-8
+        - AR=gcc-ar-8
```
Comment (hpvb) [inline .travis.yml]: probably needs CXX=g++-8 also

Comment (marxin) [reply .travis.yml]: I have the same feeling :+1:

**Comment by hpvb** [comment]
This needs something like:
```
before_install:
  - sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
  - sudo apt-get update -q
  - sudo apt-get install gcc-8 -y
```

**Comment by marxin** [comment]
Took me quite some iterations, but it's working now. Later, I would switch to gcc-8 as a default GCC compiler and leave just a legacy gcc-48 for compatibility.

**Comment by akien-mga** [comment]
Thanks, looks good.

> Later, I would switch to gcc-8 as a default GCC compiler and leave just a legacy gcc-48 for compatibility.

I'll work on that and refactor things a bit.

**Pull request merged**